### PR TITLE
[logging-too-many-args] Add functional tests for each value of ``logging-format-style``

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Refs #9118

### CODE REVIEW

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas Do you have more WIP here? Or shall we close this?

**Comment by Pierre-Sassoulas** [comment]
Creating regression tests is work too, why throw it away ? But, I don't have the fix atm no.

**Comment by DanielNoord** [comment]
> Creating regression tests is work too, why throw it away ? But, I don't have the fix atm no.

I think a draft PR with just a bunch of test cases gives a false impression to people stumbling upon the repository that something is being worked upon. We could just post the test cases in the linked issue as a started for somebody who would like to work on this and not have PRs that won't get finished in the PR overview. But perhaps that is just my personal preference. Sorry!

**Comment by Pierre-Sassoulas** [comment]
Hmm yeah, that make sense if there's only a functional test snipet that can be copy pasted from the issue. This one span multiple files, hard to 'save'. Need take over label ?

**Comment by Pierre-Sassoulas** [comment]
Or maybe we merge the wrong behavior with proper comment that it's an issue ?

**Comment by DanielNoord** [comment]
The regression tests are two lines right: https://github.com/pylint-dev/pylint/pull/9121/commits/921be718f01909d7f70f0550cceacf3c5a0c76e2

The rest is just moving files around and duplicating tests into `new` and `old` style which probably don't belong there? I'm fine with "needs take over" but I have personally never seen that lead to somebody actually picking up the PR 😅

**Comment by Pierre-Sassoulas** [comment]
All right, let's merge the refactor and copy paste the functional test snipped then.

**Comment by Pierre-Sassoulas** [comment]
All right the I was on mobile and didn't see the code. Now that I read it and the context the intent of the MR is due to this : https://github.com/pylint-dev/pylint/issues/9118#issuecomment-1749589514

It was hard to reproduce and the current test is not testing each possibility with each values of ``logging-format-style=new``. Actually the new regression test is 4 lines in 2 files and thus hard to copy paste. I think we should merge and acknowledge that there's a false positive, if someone works on it it's going to make their live easier.

**Comment by DanielNoord** [comment]
I do think we need to explain why we are mixing old and new style tests in a file that has a docstring of "Tests for logging-too-many-args using old style". That will definitely confuse us later on.

**Comment by Pierre-Sassoulas** [comment]
It's confusing me right now 😄

File: tests/functional/b/broad_exception/broad_exception_caught.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
-broad-exception-caught:14:7:14:16::Catching too general exception Exception:INFERENCE
-broad-exception-caught:20:7:20:20::Catching too general exception BaseException:INFERENCE
-broad-exception-caught:32:7:32:27::Catching too general exception CustomBroadException:INFERENCE
+broad-exception-caught:14:7:14:16::Catching too general exception Exception:INFERENCE
+broad-exception-caught:20:7:20:20::Catching too general exception BaseException:INFERENCE
+broad-exception-caught:32:7:32:27::Catching too general exception CustomBroadException:INFERENCE
```
Comment (Pierre-Sassoulas) [inline tests/functional/b/broad_exception/broad_exception_caught.txt:3]: Not sure why those two file changed !

**Review by DanielNoord — DISMISSED** [review]

**Comment by Pierre-Sassoulas** [comment]
Updated the docstring for clarity, this is ready for review now.

**Comment by Pierre-Sassoulas** [comment]
Thanks for the review and thanks for the general cleanup earlier Daniel. I did not check everything but I'm not sure we should loose hope on the "need takes over" label and close them all when they age. Some contributors are giving up really close to completion, there's some value in those MR (but we should close low value or unrecoverable one for sure).

**Comment by DanielNoord** [comment]
> Thanks for the review and thanks for the general cleanup earlier Daniel. I did not check everything but I'm not sure we should loose hope on the "need takes over" label and close them all when they age. Some contributors are giving up really close to completion, there's some value in those MR (but we should close low value or unrecoverable one for sure).

Yeah, those that were close or more recent I kept open. Some of the more controversial ones or low effort ones I closed!

**Comment by Pierre-Sassoulas** [comment]
Sorry the new comment moved the whole functional test output, had to regenerate 😅

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**